includes/class-wc-subscriptions-data-copier.php

```diff
@@ -181,6 +181,14 @@ private function set_data( $key, $value ) {
 			return;
 		}
 
+		// The WC_Order setter for these keys will expect an array of values, return early if the value is not an array.
+		if (
+			in_array( $key, [ '_shipping_address', '_shipping', '_billing_address', '_billing' ], true )
```

Comment:
Are these setters in WC core? And … should they be more robust (no fatal if passed non-array)?

Adding this check is a good layer of protection, but makes me wonder if we're protecting some code that needs to be more robust.

Reply:
> Are these setters in WC core? 

Yes, I've updated the description with a list of the WC_Order setters we're preventing from being called with unsupported values.

>And … should they be more robust (no fatal if passed non-array)?

It seems that way. As per our Slack discussion, we'll be opening an issue in WC for this.